worlds/doom_ii/__init__.py

```diff
@@ -0,0 +1,268 @@
+
+        # Items
+        for item_id, item in Items.item_table.items():
+            if item["doom_type"] == DOOM_TYPE_LEVEL_COMPLETE:
+                continue # We'll fill it manually later
+
+            if item["doom_type"] == DOOM_TYPE_COMPUTER_AREA_MAP and start_with_computer_area_maps:
+                continue # We'll fill it manually, and we will put fillers in place
+
+            if item["episode"] != -1 and not self.included_episodes[item["episode"] - 1]:
+                continue
+
+            count = item["count"] if item["name"] not in self.starting_level_for_episode else item["count"] - 1
+            itempool += [self.create_item(item["name"]) for _ in range(count)]
+
+        # Place end level items in locked locations
+        for map_name in Maps.map_names:
+            loc_name = map_name + " - Exit"
+            item_name = map_name + " - Complete"
+
+            if loc_name not in self.location_name_to_id:
+                continue
+
+            if item_name not in self.item_name_to_id:
+                continue
+
+            loc = Locations.location_table[self.location_name_to_id[loc_name]]
+            if not self.included_episodes[loc["episode"] - 1]:
+                continue
+
+            self.multiworld.get_location(loc_name, self.player).place_locked_item(self.create_item(item_name))
+            self.location_count -= 1
+
+        # Give starting levels right away
+        for i in range(len(self.starting_level_for_episode)):
+            if self.included_episodes[i]:
+                self.multiworld.push_precollected(self.create_item(self.starting_level_for_episode[i]))
+        
+        # Give Computer area maps if option selected
+        if getattr(self.multiworld, "start_with_computer_area_maps")[self.player].value:
+            for item_id, item_dict in Items.item_table.items():
+                item_episode = item_dict["episode"]
+                if item_episode > 0:
+                    if item_dict["doom_type"] == DOOM_TYPE_COMPUTER_AREA_MAP and self.included_episodes[item_episode - 1]:
+                        self.multiworld.push_precollected(self.create_item(item_dict["name"]))
+
+        # Fill the rest starting with powerups, then fillers
+        self.create_ratioed_items("Armor", itempool)
+        self.create_ratioed_items("Mega Armor", itempool)
+        self.create_ratioed_items("Berserk", itempool)
+        self.create_ratioed_items("Invulnerability", itempool)
+        self.create_ratioed_items("Partial invisibility", itempool)
+        self.create_ratioed_items("Supercharge", itempool)
+
+        while len(itempool) < self.location_count:
+            itempool.append(self.create_item(self.get_filler_item_name()))
+
+        # add itempool to multiworld
+        self.multiworld.itempool += itempool
+
+    def get_filler_item_name(self):
+        return self.multiworld.random.choice([
```

Comment:
```suggestion
        return self.random.choice([
```

Reply:
What action should I take here? If the new code won't work, the DOOM II community will be unable to play latest until it's released. I rather use older code that is backward compatible.

Reply:
Probably leave it as multiworld for now then.